tests/MenderAPI/auth_v2.py

```diff
@@ -56,21 +56,19 @@ def get_devices_status(self, status=None, expected_devices=1):
             time.sleep(sleeptime)
             # Linear backoff
             sleeptime += 5
-            try:
-                logger.info("getting all devices from :%s" % (device_status_path))
-                devices = requests_retry().get(device_status_path, headers=self.auth.get_auth_token(), verify=False)
-                assert devices.status_code == requests.status_codes.codes.ok
-                assert len(devices.json()) == expected_devices
+            logger.info("getting all devices from: %s" % (device_status_path))
+            devices = requests_retry().get(device_status_path, headers=self.auth.get_auth_token(), verify=False)
+            if devices.status_code == requests.status_codes.codes.ok and len(devices.json()) == expected_devices:
                 break
-            except AssertionError:
+            else:
                 if devices is not None and getattr(devices, "text"):
                     logger.info("fail to get devices (payload: %s), will try for at least %d more seconds"
                                 % (devices.text, starttime + max_wait - time.time()))
                 else:
                     logger.info("failed to get devices, will try for at least %d more seconds"
                                 % (starttime + max_wait - time.time()))
-                continue
-        else:
+
+        if max_wait >= time.time():
```

Comment:
I think this will always be false. `max_wait` is only `60*60`, so you need to add `starttime` like in the other calculations.

Reply:
Of course it is wrong. Thanks

Reply:
And technically speaking, it would still be wrong unless I substract the last `sleeptime`. For instance when the request succeeds in the very last iteration. So I think I'll go simple defining a flag for this.